### ISSUE TITLE
frame number in references is wrong, when using subsections

### ISSUE THREAD

**Issue by josephwright** [josephwright/beamer#117]
Hi,

if I use sections and subsections, then the frame number inserted by `\ref`-erencing to the label of another frame is wrong. The problem is that there is a counter called `subsectionslide` that counts the number of slides since the last \subsection; the bug is that it is incremented by `\refstepcounter`, so this is what is (wrongly) associated to the label. I attach a test file showing the problem; and the patch (for `beamer` 3.10).

a.

ps: this was posted as bug 2089504 in sourceforge, together with a patch, but the patch was never applied; I hope this time it is.
````
\documentclass{beamer}

\usetheme[secheader]{Boadilla}

\makeatletter
\def\SHOWCURRENTLABEL{\@currentlabel}
\makeatother

\begin{document}
\section{sec1}

\begin{frame}
  the following labels are almost all wrong
  
  see frame \ref{lastframe} (should be frame 7)

  see frame \ref{lastframealt} (should be frame 7)

  see frame \ref{lastbut1frame} (should be frame 6)

  see frame \ref{lastbut1framealt} (should be frame 6)

  They all work if you delete all sections and subsections from the file.
\end{frame}

\begin{frame}
  frame  \insertframenumber
\end{frame}

\section{sec2}

\begin{frame}
  frame  \insertframenumber
\end{frame}

\section{sec3}\label{sec3}

\begin{frame}
  frame  \insertframenumber
\end{frame}
\begin{frame}
  frame  \insertframenumber
  \pause
  bis
\end{frame}

\subsection{sec3sub1}\label{sec3sub1}

\begin{frame}
  \label{lastbut1frame}
  \label<1>{lastbut1framealt}
  frame  \insertframenumber

  this frame has labels but no overlays.

  I cant seem to find a way to get the correct frame number
  out of the label!
\end{frame}

\begin{frame}
  \label{lastframe}
  It is quite interesting to see how the internal
  value of the current label
  \begin{center}
    \SHOWCURRENTLABEL
  \end{center}
  changes while viewing this frame's slides.
  Note how the first value shown is also the value
  given to the label ``lastframe'' (and is wrong, since this
  is frame \insertframenumber)

  \pause  
  
  1
  
  \pause   \label<2>{lastframealt} 
  the label inserted in overlay 2 (or following) is the only one that works

  
  2
  
  \pause  
  3
 
\end{frame}
\end{document}
````
Patch
````
--- /usr/share/texmf/tex/latex/beamer/base/beamerbaseframe.sty.orig	2008-09-12 17:45:17.000000000 +0200
+++ /usr/share/texmf/tex/latex/beamer/base/beamerbaseframe.sty	2008-09-12 17:45:41.000000000 +0200
@@ -55,7 +58,7 @@
   \fi%
   \global\c@beamerpauses=1\relax%
   \expandafter\beamer@ifempty\expandafter{\beamer@framestartpage}{%
-    \refstepcounter{subsectionslide}%
+    \stepcounter{subsectionslide}%
     \xdef\beamer@framestartpage{\the\c@page}% only first time
   }{\clearpage\beamer@notesactions}% cleanup from previous slide
   \hypersetup{pdfpagetransition=R}%
````
----------------------------------------
- Bitbucket: https://bitbucket.org/rivanvx/beamer/issue/117
- Originally reported by: A Mennucci
- Originally created at:  2011-02-26

**Comment by josephwright** [comment]
_From A Mennucci on  2011-02-26_

For your convenience, here is an example short code (but I recommend at looking at the attached example)
````
\begin{frame}
  see frame \ref{HIframe}
\end{frame}
\subsection{sec3sub1}
\begin{frame}\label{HIframe}
 HI
\end{frame}
````
If you use `\subsection`s in the document, then the numbers shown by `\ref` are wrong (see attachment example)

a.

**Comment by josephwright** [comment]
Fixed by c2582a30843be64ddded814306f2c7123979a71a
